### PR TITLE
Improve `argilla` integration

### PR DESCRIPTION
**Pull request by alvarobartt**
## Description

This PR improves the integration with `argilla` to not just make the code simpler, but also to make it easier to understand and go through for new Argilla users.

Besides that, we've included some args that map to the expected column names within the dataset, in case the user wants to modify those e.g. use `responses` instead of `generations`, or the other way around, so that the column names `rationale`, `ratings`, and `generations` can be mapped to different names.

Additionally, the pinned version of Argilla in `pyproject.toml` has been fixed as the minimum required version as of the `metadata_properties` addition is now v1.18.0 instead of v1.16.0; and the `README.md` has been slightly curated and some typos have been fixed too.

### CODE REVIEW

File: src/distilabel/dataset.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 from distilabel.utils.imports import _ARGILLA_AVAILABLE
 
 if _ARGILLA_AVAILABLE:
-    import argilla as rg
+    pass
```
Comment (alvarobartt) [inline src/distilabel/dataset.py:22]: To be reverted

```suggestion
    import argilla as rg
```

File: src/distilabel/tasks/preference/base.py [hunk 2]
```diff
@@ -44,129 +47,158 @@ def output_args_names(self) -> List[str]:
...
-
-    def to_argilla_questions(
-        self, dataset_row: Dict[str, Any]
-    ) -> List["AllowedQuestionTypes"]:
-        """Converts a dataset row to a list of Argilla `AllowedQuestionTypes`."""
+    def to_argilla_dataset(
+        self,
+        dataset_row: Dict[str, Any],
+        generations_column: Optional[str] = "generations",
+        responses_values: Optional[List[int]] = None,
```
Comment (alvarobartt) [inline src/distilabel/tasks/preference/base.py:54]:
```suggestion
        generations_ratings: Optional[List[int]] = None,
```

File: src/distilabel/tasks/argilla_utils.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+from typing import TYPE_CHECKING, Any, Dict, List
+
+from distilabel.utils.imports import _ARGILLA_AVAILABLE
+
+if _ARGILLA_AVAILABLE:
+    import argilla as rg
+
+    if TYPE_CHECKING:
+        from argilla.client.feedback.schemas.types import AllowedFieldTypes
```
Comment (gabrielmbmb) [inline src/distilabel/tasks/argilla_utils.py:23]: We can unnest this if you want

File: src/distilabel/tasks/preference/base.py [hunk 4]
```diff
@@ -12,18 +12,21 @@
...
+from distilabel.tasks.argilla_utils import infer_fields_from_dataset_row
 from distilabel.tasks.base import Task
+from distilabel.utils.imports import _ARGILLA_AVAILABLE
+
+if _ARGILLA_AVAILABLE:
+    import argilla as rg
+
+    if TYPE_CHECKING:
+        from argilla.client.feedback.dataset.local.dataset import FeedbackDataset
+        from argilla.client.feedback.schemas.records import FeedbackRecord
```
Comment (gabrielmbmb) [inline src/distilabel/tasks/preference/base.py:28]: we can unnest this

File: src/distilabel/tasks/preference/base.py [hunk 5]
```diff
@@ -44,129 +47,158 @@ def output_args_names(self) -> List[str]:
...
+        fields = infer_fields_from_dataset_row(
+            field_names=self.input_args_names, dataset_row=dataset_row
+        )
+        # Then we add the questions, which cannot be easily inferred in this case,
+        # because those depend neither on the outputs nor on the inputs, but in a combination
+        # of both, since the questions will be formulated using the inputs, but assigned to the
+        # outputs.
+        if generations_column is None or generations_column not in dataset_row:
+            raise ValueError(
+                f"The generations column {generations_column} is not present in the dataset row."
```
Comment (gabrielmbmb) [inline src/distilabel/tasks/preference/base.py:69]:
```suggestion
                f"The generations column '{generations_column}' is not present in the dataset row."
```

File: src/distilabel/tasks/preference/base.py [hunk 6]
```diff
@@ -44,129 +47,158 @@ def output_args_names(self) -> List[str]:
...
+            )
+        if ratings_column is None or ratings_column not in dataset_row:
+            raise ValueError(
+                f"The ratings column {ratings_column} is not present in the dataset row."
```
Comment (gabrielmbmb) [inline src/distilabel/tasks/preference/base.py:73]:
```suggestion
                f"The ratings column '{ratings_column}' is not present in the dataset row."
```

File: src/distilabel/tasks/preference/base.py [hunk 7]
```diff
@@ -44,129 +47,158 @@ def output_args_names(self) -> List[str]:
...
+            )
+        if rationale_column is None or rationale_column not in dataset_row:
+            raise ValueError(
+                f"The rationale column {rationale_column} is not present in the dataset row."
```
Comment (gabrielmbmb) [inline src/distilabel/tasks/preference/base.py:77]:
```suggestion
                f"The rationale column '{rationale_column}' is not present in the dataset row."
```

**Review by gabrielmbmb — APPROVED** [review]

**Pull request merged**